buildpack.md

```diff
@@ -567,25 +640,31 @@ It MUST NOT be used for environment variables that are not defined in this speci
 
 ### Provided by the Buildpacks
 
-During the build phase, buildpacks MAY write environment variable files to `<cache>/<layer>/env/` directories.
+During the build phase, buildpacks MAY write environment variable files to `<layers>/<layer>/env/`,  `<layers>/<layer>/env.build/`, and `<layers>/<layer>/env.launch/` directories.
+
+For each `<layers>/<layer>/` designated as a build layer, for each file written to `<layers>/<layer>/env/` or `<layers>/<layer>/env.build/` by `/bin/build`, the lifecycle MUST modify an environment variable in subsequent executions of `/bin/build`.
+
+For each file written to `<layers>/<layer>/env.launch/` by `/bin/build`, the lifecycle MUST modify an environment variable when the OCI image is launched.
 
-For each file written to `<cache>/<layer>/env/` by `/bin/build`, the lifecycle MUST modify an environment variable in subsequent executions of `/bin/build`.
 The lifecycle MUST set the name of the environment variable to the name of the file up to the first period (`.`) or to the end of the name if no periods are present.
 
-If the environment variable has no period-delimited suffix, then the value of the environment variable MUST be a concatenation of the file contents and the contents of other identically named files in other `<cache>/<layer>/env/` directories delimited by the OS path list separator.
+If the environment variable file name has no period-delimited suffix, then the value of the environment variable MUST be a concatenation of the file contents and the contents of other identically named files delimited by the OS path list separator.
 Within that environment variable value,
 - Later buildpacks' environment variable file contents MUST precede earlier buildpacks' environment variable file contents.
-- Environment variable file contents originating from the same buildpack MUST be sorted alphabetically ascending by associated layer name.
+- Environment variable file contents originating from the same buildpack MUST be sorted alphabetically descending by associated layer name.
+- Environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/` precede file contents in `<layers>/<layer>/env/`.
 
-If the environment variable file name ends in `.append`, then the value of the environment variable MUST be a concatenation of the file contents and the contents of other identically named files in other `<cache>/<layer>/env/` directories without any delimitation.
+If the environment variable file name ends in `.append`, then the value of the environment variable MUST be a concatenation of the file contents and the contents of other identically named files without any delimitation.
```

Comment:
When this says "without any delimitation" what did we envision being the use-case for that?  I understand the path separated version, but I'd somehow envisioned this to be space delimited.

Reply:
You could add space or newline delimiters with this feature, as long as you're willing to have an extra space or newline at the end. Maybe this interface could be improved?

Reply:
I'm not horribly opposed to this (I even noticed that when I implemented `libbuildpack` previously there's a great big doc note around this method that you're on your own with delimiters.  I'm using this behavior similarly to appending to a path where you always append with `:<path>` and the whole system is prepared to ignore that first empty value.  this mostly makes sense with `JAVA_OPTS` where having `<space>-Dfoo=bar<space>-Dalpha=bravo` is perfectly reasonable.

Reply:
For paths, use the version without `.append` -- that appends the correct OS path separator automatically.